pkg/awsutils/awsutils.go

```diff
@@ -419,28 +415,33 @@ func New(useCustomNetworking, disableLeakedENICleanup, v4Enabled, v6Enabled bool
 	}
 	cache.region = region
 	log.Debugf("Discovered region: %s", cache.region)
-
 	cache.useCustomNetworking = useCustomNetworking
 	log.Infof("Custom networking enabled %v", cache.useCustomNetworking)
-
 	cache.v4Enabled = v4Enabled
 	cache.v6Enabled = v6Enabled
 
 	awsCfg := aws.NewConfig().WithRegion(region)
 	sess = sess.Copy(awsCfg)
-
 	ec2SVC := ec2wrapper.New(sess)
 	cache.ec2SVC = ec2SVC
 	err = cache.initWithEC2Metadata(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	// Setup Event Recorder, which creates a Kubernetes event when IAM permissions are missing
+	eventRecorder, err := eventrecorder.New(cachedK8SClient)
+	if err != nil {
+		log.Errorf("Failed to create event recorder: %v", err)
+		return nil, errors.Wrapf(err, "failed to create event recorder")
+	}
+	eventRecorder.HostID = cache.instanceID
+	cache.eventRecorder = eventRecorder
```

Comment:
nit: we are adding `eventRecorder` to `EC2InstanceMetadataCache` which is not a metadata property. It would have been good if we had something like - `var log = logger.Get()` for eventRecorder...

Reply:
Yeah, `EC2InstanceMetadataCache` is the only user today, but it would be better long-term if `EventRecorder` created itself and others could just call `Get`. I'll try that out tomorrow

Reply:
This has been refactored in the latest revision